src/main/java/de/tum/in/www1/artemis/service/programming/RepositoryService.java

```diff
@@ -116,20 +116,23 @@ public Map<String, FileType> getFiles(Repository repository) {
      * @throws IOException     If an I/O error occurs during the file content retrieval process. This could be due to issues with file access, network problems, etc.
      * @throws GitAPIException If an error occurs while interacting with the Git repository. This could be due to issues with repository access, invalid commit ids, etc.
      */
-    public Map<String, String> getFilesContentAtCommit(ProgrammingExercise programmingExercise, String commitId, RepositoryType repositoryType,
+    public Map<String, String> getFilesContentAtCommit(ProgrammingExercise programmingExercise, String commitId, Optional<RepositoryType> repositoryType,
             ProgrammingExerciseParticipation participation) throws IOException, GitAPIException {
+
+        final boolean isTestRepository = repositoryType.equals(Optional.of(RepositoryType.TESTS));
         // Check if local VCS is active
         if (profileService.isLocalVcsActive()) {
             log.debug("Using local VCS for getting files at commit {} for participation {}", commitId, participation.getId());
             // If local VCS is active, operate directly on the bare repository
-            var repoUri = repositoryType == RepositoryType.TESTS ? programmingExercise.getVcsTestRepositoryUri() : participation.getVcsRepositoryUri();
+            var repoUri = isTestRepository ? programmingExercise.getVcsTestRepositoryUri() : participation.getVcsRepositoryUri();
             Repository repository = gitService.getBareRepository(repoUri);
             return getFilesContentFromBareRepository(repository, commitId);
         }
         else {
             log.debug("Checking out repo to get files at commit {} for participation {}", commitId, participation.getId());
+
             Repository repository;
-            if (repositoryType == RepositoryType.TESTS) {
+            if (isTestRepository) {
                 repository = gitService.checkoutRepositoryAtCommit(programmingExercise.getVcsTestRepositoryUri(), commitId, true);
```

Comment:
This looks a bit strange. Can you maybe check before calling this method if the optional is empty and then call this only with the RepositoryType present?

Reply:
There is no really clear answer here. The (as of now) 3 usages make different assumptions:

- `ProgrammingExerciseParticipationResource::getParticipationRepositoryFiles` assumes the called method handles `null`
- `ProgrammingExerciseParticipationResource::getParticipationRepositoryFilesForCommitsDetailsView` handles the null or now empty case itself
- `RepositoryProgrammingExerciseParticipationResource::getFilesAtCommit` uses `RepositoryResource::executeAndCheckForExceptions` to kind of handle the null case

I do not see a clear consensus here how this method should used. The JavaDoc in question also does not state how `repositoryType` being `null` is handled. As of now it assumes that `repositoryType` being `null` means "not `RepositoryType.TESTS`".

Reply:
I think we could simplify this method in general

```
public Map<String, String> getFilesContentAtCommit(ProgrammingExercise programmingExercise, String commitId, Optional<RepositoryType> repositoryType,
        ProgrammingExerciseParticipation participation) throws IOException, GitAPIException {

    boolean isTestRepository = repositoryType.equals(Optional.of(RepositoryType.TESTS));
    String repoUri = isTestRepository ? programmingExercise.getVcsTestRepositoryUri() : participation.getVcsRepositoryUri();

    if (profileService.isLocalVcsActive()) {
        log.debug("Using local VCS for getting files at commit {} for participation {}", commitId, participation.getId());
        var repository = gitService.getBareRepository(repoUri);
        return getFilesContentFromBareRepository(repository, commitId);
    } else {
        log.debug("Checking out repo to get files at commit {} for participation {}", commitId, participation.getId());
        var repository = gitService.checkoutRepositoryAtCommit(repoUri, commitId, true);
        Map<String, String> filesWithContent = getFilesContentFromWorkingCopy(repository);
        gitService.switchBackToDefaultBranchHead(repository);
        return filesWithContent;
    }
}
```

Reply:
Done.